server/metrics/index.js

```diff
@@ -0,0 +1,37 @@
+const Prometheus = require('prom-client')
+const { buildPath } = require('../lib/util/logger')
+
+Prometheus.collectDefaultMetrics()
+
+const resolverTimingMetric = new Prometheus.Gauge({
+  name: 'resolver_timing_ms',
+  help: 'Time period between request and response in milliseconds',
+  labelNames: ['type', 'name']
+})
+
+const requestsResolvedMetric = new Prometheus.Counter({
+  name: 'requests_resolved',
+  help: 'Number of requests resolved by server',
+  labelNames: ['type', 'path']
+})
+
+exports.getMetrics = async (req, res) => {
+  res.set('Content-Type', Prometheus.register.contentType)
+  res.end(Prometheus.register.metrics())
+}
+
+exports.updateResolverMetrics = (resolverInfo, responseTime) => {
+  const {
+    operation: {operation: resolverMappingType},
```

Comment:
I couldn't understand where this variable comes from. Maybe too advanced JS for me :laughing:

Reply:
@aliok Two what's happening here is a sort of nested destructuring of properties from the resolverInfo Object.

You've probably unfamiliar with the names of the fields because it's internal stuff from the GraphQL engine passed as the `info` arg into the resolver function. You can see this up in the code where `updateResolverMetrics` is called from 😄